server/lobbyconnection.py

```diff
@@ -124,7 +124,6 @@ async def abort(self, logspam=""):
             )
         if self.game_connection:
             await self.game_connection.abort()
-            self.game_connection = None
```

Comment:
If a client sends a `GameState ended` message and immediately afterwards a wrong game result, it feels like it could happen that the first message triggers the `game_connection.abort()`, then the thread sleeps before the game_connection is set to `None` and then the second messages' result still gets set.
I think that's very unlikely, given that the result thread will also `await` something, but at this point it feels like we're relying on asyncio internals a bit - what if there are a lot more `await`s in the `game_connection.abort()` than in the result processing?

So maybe some flag should still be set immediately here in `lobbyconnection.py`, or wherever `GameState ended` is processed. (Or unset the `game_connection` reference, but then it's slightly awkward to call `abort()` afterwards.)

Whether you think that's an issue or not, maybe a test checking that when you send `GameState ended` and then immediately a result, the result doesn't get saved is a good idea?

Reply:
I'll add some game results to the test